### ISSUE TITLE
Possible race condition in Keepalived/HAProxy/OpenBFDd

### ISSUE THREAD

**Issue by vt0r** [bloomberg/chef-bach#847]
Getting the below error when trying to run `tests/automated_install.sh` to do a new VM cluster build. Logging into the VM after this error is encountered, you can ping the VIP and connect to MySQL on it using the vaulted root password, hence the title.

```
Recipe: bcpc::powerdns
  * chef_vault_secret[mysql-pdns] action nothing (skipped due to action :nothing)
  * mysql_database[pdns] action create
    
    ================================================================================
    Error executing action `create` on resource 'mysql_database[pdns]'
    ================================================================================
    
    Mysql2::Error
    -------------
    Can't connect to MySQL server on '10.0.100.5' (113)
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:105:in `new'
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:105:in `test_client'
    /var/chef/cache/cookbooks/database/libraries/provider_database_mysql.rb:37:in `block in <class:Mysql>'
    
    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/bcpc/recipes/powerdns.rb
    
     90:   mysql_database node['bcpc']['pdns_dbname'] do
     91:     connection lazy { mysql_connection_info.call }
     92:     notifies :run, 'execute[install-pdns-schema]', :immediately
     93:   end
     94: 
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/bcpc/recipes/powerdns.rb:90:in `from_file'
    
    mysql_database("pdns") do
      provider Chef::Provider::Database::Mysql
      action :create
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      database_name "pdns"
      declared_type :mysql_database
      cookbook_name "bcpc"
      recipe_name "powerdns"
      connection #<Chef::DelayedEvaluator:0x00000002e64cc0@/var/chef/cache/cookbooks/bcpc/recipes/powerdns.rb:91>
    end
    
    Platform:
    ---------
    x86_64-linux
    
Recipe: bcpc::mysql
  * service[mysql] action reload
    - reload service service[mysql]
Recipe: bcpc::keepalived
  * service[keepalived] action restart
    - restart service service[keepalived]

Running handlers:
  - GraphiteReporting
Running handlers complete
Chef Client failed. 114 resources updated in 04 minutes 40 seconds
bundler: failed to load command: ./cluster_assign_roles.rb (./cluster_assign_roles.rb)
RuntimeError: bcpc-vm1.bcpc.example.com: Got failed status 1 from chef run!
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:168:in `chef_node_with_runlist'
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:216:in `block in install_bootstrap'
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:214:in `each'
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:214:in `install_bootstrap'
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:478:in `car_cli'
  /home/vagrant/chef-bcpc/cluster_assign_roles.rb:486:in `<top (required)>'
```

**Comment by pu239ppy** [comment]
@cbaenziger @vt0r Did not encounter this in my last build of master, was this fixed recently

**Comment by cbaenziger** [comment]
I've had this not catch some of my runs. The "joys" of this being a rance only I think.

**Comment by pu239ppy** [comment]
This happened consistently for some time, and then stopped, I don't remember this being a once-in-a-while issue, I believe that I saw this error scroll by at some point, but it did not stop the show

**Comment by pu239ppy** [comment]
Nevermind happened again

**Comment by pu239ppy** [comment]
As this happens at the CAR Bootstrap stage, a hack around might be to just rerun CAR Bootstrap

**Comment by pu239ppy** [comment]
````bash
printf "#### Chef machine bcpc-vms with Bootstrap\n"
set +e
vagrant ssh -c "cd chef-bcpc; ./cluster-assign-roles.sh $ENVIRONMENT Bootstrap"
set -e
vagrant ssh -c "cd chef-bcpc; ./cluster-assign-roles.sh $ENVIRONMENT Bootstrap"
printf "Snapshotting Post-Bootstrap\n"
for i in bootstrap vm1 vm2 vm3; do
  [[ $(vboxmanage snapshot bcpc-$i list --machinereadable | grep -q 'Post-Bootstrap') ]] && \
    VBoxManage snapshot bcpc-$i take Post-Bootstrap &
done
wait && printf "Done Snapshotting\n"
````

**Comment by vt0r** [comment]
Fixed by #882 adding enough time to force a winner in the race.
